internal/graph/cached_resolver.go

```diff
@@ -109,9 +110,10 @@ func NewCachedCheckResolver(opts ...CachedCheckResolverOpt) *CachedCheckResolver
 
 	if checker.cache == nil {
 		checker.allocatedCache = true
-		checker.cache = ccache.New(
-			ccache.Configure[*ResolveCheckResponse]().MaxSize(checker.maxCacheSize),
-		)
+		var optsCache []storage.InMemoryLRUCacheOpt[*ResolveCheckResponse]
+		optsCache = append(optsCache, storage.WithMaxCacheSize[*ResolveCheckResponse](checker.maxCacheSize))
+		optsCache = append(optsCache, storage.WithDefaultTTL[*ResolveCheckResponse](checker.cacheTTL))
+		checker.cache = storage.NewInMemoryLRUCache[*ResolveCheckResponse](optsCache...)
```

Comment:
The `append` function performs various checks that are not necessary in a scenario where we know the upper-bound and items of the slice. It is generally more performant to use direct initialization, or initialize the slice with a length (using `make`) equal to the upper-bound and use direct index access to assign values.
```suggestion
		optsCache := []storage.InMemoryLRUCacheOpt[*ResolveCheckResponse]{
		    storage.WithMaxCacheSize[*ResolveCheckResponse](checker.maxCacheSize),
		    storage.WithDefaultTTL[*ResolveCheckResponse](checker.cacheTTL)
		}
		checker.cache = storage.NewInMemoryLRUCache[*ResolveCheckResponse](optsCache...)
```